### PR TITLE
helm chart and code enhancements

### PR DESCRIPTION
**Pull request by neillturner**
add helm chart
add support for github app authentication
change worflowCache to only process when not an error from ListWorkflows call.

It would be great if you could host the helm chart.
the easiest way is to host on github itself  
see https://dev.to/jamiemagee/how-to-host-your-helm-chart-repository-on-github-3kd
just need to create a helm_charts repo with a gh-pages branch or even just a gh-pages branch of your repo. 

This change can be used from:
```
 helm chart repo: https://neillturner.github.io/helm-charts 
 chart: github-actions-exporter
 version: "0.1.4"
```
see sample flux V2 at `https://github.com/neillturner/flux-examples/tree/main/github-actions-exporter`

Hope this is helpful. 

Thanks

### CODE REVIEW

**Comment by Labbs** [comment]
Hello,

Thanks for the feature @neillturner @oneubauer. Just need fix the conflict after and I merge.

**Comment by Labbs** [comment]
https://github.com/ec2dream/github-actions-exporter/pull/1

**Comment by peimanja** [comment]
Any updates on this PR? Interested trying out this exporter and this would make it easier since we want to use the Github app and need the helm chart as well

**Comment by neillturner** [comment]
fixed conflicts

**Comment by neillturner** [comment]
had a problem with head_branch giving a memory segmentation error for some reason
to workaround had to specify: 
 EXPORT_FIELDS: "repo,id,node_id,head_sha,run_number,workflow_id,workflow,event,status"
if you update from "github.com/google/go-github/v33/github"

**Comment by callum-tait-pbx** [comment]
I'd love to see this merged @Labbs as I'd love to test this to resolve some of the observability issues the GitHub Actions platform has i.e. literally 0 built in observability

We are a Helm shop so a chart for this is an absolute requirement for me and my environment and I can't be the only one! :D

File: README.md [hunk 1]
```diff
@@ -166,8 +175,28 @@ Example:
 github_workflow_usage_seconds{id="2862037",name="Create Release",node_id="MDg6V29ya2Zsb3cyODYyMDM3",repo="xxx/xxx",state="active",os="UBUNTU"} 706.609
 ```
 
+## Setting up authentication with GitHub API
+
+There are two ways for github-actions-exporter to authenticate with the GitHub API (only 1 can be configured at a time however):
+
+1. Using a GitHub App (not supported when you use Github Enterprise )
```
Comment (toast-gear) [inline README.md:182]: I don't think is quite right? GitHub Enterprise is just a subscription https://github.com/pricing. Do you mean:

```
1. Using a GitHub App (not supported at the GitHub Enterprise management hierarchy level)
```

File: README.md [hunk 2]
```diff
@@ -166,8 +175,28 @@ Example:
...
+
+### Deploying using GitHub App Authentication
+
+You can create a GitHub App for either your account or any organization. If you want to create a GitHub App for your account, open the following link to the creation page, enter any unique name in the "GitHub App name" field, and hit the "Create GitHub App" button at the bottom of the page.
+
+- [Create GitHub Apps on your account](https://github.com/settings/apps/new?url=http://github.com/actions-runner-controller/actions-runner-controller&webhook_active=false&public=false&administration=write&actions=read)
+
+If you want to create a GitHub App for your organization, replace the `:org` part of the following URL with your organization name before opening it. Then enter any unique name in the "GitHub App name" field, and hit the "Create GitHub App" button at the bottom of the page to create a GitHub App.
+
+- [Create GitHub Apps on your organization](https://github.com/organizations/:org/settings/apps/new?url=http://github.com/actions-runner-controller/actions-runner-controller&webhook_active=false&public=false&administration=write&organization_self_hosted_runners=write&actions=read)
```
Comment (toast-gear) [inline README.md:197]: These URLs are for a different project

File: README.md [hunk 3]
```diff
@@ -166,8 +175,28 @@ Example:
...
+
+If you are deploying the solution for a GitHub Enterprise Server environment you are able to [configure your rate limiting settings](https://docs.github.com/en/enterprise-server@3.0/admin/configuration/configuring-rate-limits) making the main benefit irrelevant. If you're deploying the solution for a GitHub Enterprise Cloud or regular GitHub environment and you run into rate limiting issues, consider deploying the solution using the GitHub App authentication method instead.
+
+### Deploying using GitHub App Authentication
+
+You can create a GitHub App for either your account or any organization. If you want to create a GitHub App for your account, open the following link to the creation page, enter any unique name in the "GitHub App name" field, and hit the "Create GitHub App" button at the bottom of the page.
+
+- [Create GitHub Apps on your account](https://github.com/settings/apps/new?url=http://github.com/actions-runner-controller/actions-runner-controller&webhook_active=false&public=false&administration=write&actions=read)
```
Comment (toast-gear) [inline README.md:193]: This URL is for a different project

File: pkg/config/config.go [hunk 4]
```diff
@@ -3,22 +3,47 @@ package config
...
 
 var (
+	// Github - github configuration
 	Github struct {
-		Token         string
-		Refresh       int64
-		Repositories  cli.StringSlice
-		Organizations cli.StringSlice
-		ApiUrl        string
+		AppID             int64  `split_words:"true"`
```
Comment (Labbs) [inline pkg/config/config.go:8]: you can use cli.Int64Slice, the cli lib split automatly the string to int64 slice. not mandatory

**Review by Labbs — APPROVED** [review]

**Comment by Labbs** [comment]
> had a problem with head_branch giving a memory segmentation error for some reason
> to workaround had to specify:
> EXPORT_FIELDS: "repo,id,node_id,head_sha,run_number,workflow_id,workflow,event,status"
> if you update from "github.com/google/go-github/v33/github"
> to latest which is "github.com/google/go-github/v37/github"
> then get extra fields like name which is very useful. -)

You can merge. I change the version of the lib after and start a new release.

**Pull request merged**